src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -356,5 +438,76 @@ public void WriteTo(TextWriter writer, HtmlEncoder encoder)
 
             _postElement?.WriteTo(writer, encoder);
         }
+
+        private void CopyAttributesTo(IHtmlContentBuilder destination)
+        {
+            StringWriter stringWriter = null;
+
+            // Perf: Avoid allocating enumerator
+            for (var i = 0; i < (_attributes?.Count ?? 0); i++)
+            {
+                var attribute = _attributes[i];
+                destination.AppendHtml(" ");
+                destination.AppendHtml(attribute.Name);
+
+                if (attribute.Minimized)
+                {
+                    continue;
+                }
+
+                destination.AppendHtml("=\"");
+                var value = attribute.Value;
+                var htmlContent = value as IHtmlContent;
```

Comment:
Is it impossible for `_attributes` to contain an `IHtmlContentBuilder`. Unsure this is the Right Thing:tm: in the `MoveTo()` case if not.